Updated doc to be more to date with v1.9 (#6125)
dbt-labs/docs.getdbt.com: website/docs/reference/resource-properties/versions.md

- Removing or modifying one of the `constraints` on an existing column (dbt v1.6 or higher)
- Changing unversioned, contracted models. 
  - dbt also warns if a model has or had a contract but isn't versioned

> - Removing a contracted model constitutes a breaking change. A model is considered 'removed' if it is deleted, renamed, or disabled:

Reviewer: adding the 'config:enabled bit since it was mentioned in the [issue](https://github.com/dbt-labs/docs.getdbt.com/issues/5574#issue-2321803112)
```suggestion
- Removing a contracted model is a breaking change. A model is considered 'removed' if it's deleted, renamed, or disabled (using the `config: enabled`):
```

>   - When you remove a versioned model, you will receive an error.

Reviewer: ```suggestion
  - Removing a versioned model causes an error.
```
